src/components/dialog/dialog.scss

```diff
@@ -105,6 +105,7 @@ $dialog-sizes: (
 
   .carbon-heading__header {
     margin-bottom: 0;
+    padding-bottom: 15px;
```

Comment:
Why not adjust the padding on the `Heading` component rather than on the Dialog's version of it?

Reply:
This is because as a Header on a page, the spacing was required to offset the gap on the top, and in a dialog, it is smaller, so the padding needs to be reduced, but this is specific in a dialog.

Reply:
OK - would have preferred to avoid contextual minor changes like this if possible as it makes the components more brittle to change, but if it is the way it has to be ✅

Reply:
If you look at the way display flex and display grid work they both change the layout of the children based on context provided by the immediate parent, so I think this is a reasonable thing to do but crucially just for layout styles.